all_test.go

```diff
@@ -44,23 +44,26 @@ var hashHeader = regexp.MustCompile(`^# Copyright 20\d\d Google LLC
 #
 # Licensed under the Apache License, Version 2\.0 \(the "License"\);`)
 
-var noHeaderRequiredFiles = []string{".github/CODEOWNERS", "go.sum", "go.mod", ".gitignore", "LICENSE", "renovate.json"}
-
 func TestHeaders(t *testing.T) {
+	excludeDirs, excludeFiles := loadHeaderTestExclusions()
 	sfs := os.DirFS(".")
 	fs.WalkDir(sfs, ".", func(path string, d fs.DirEntry, err error) error {
 		if err != nil {
 			return err
 		}
 		if d.IsDir() {
-			if d.Name() == "testdata" || d.Name() == ".git" {
-				return fs.SkipDir
+			for _, excludeDir := range excludeDirs {
+				if d.Name() == excludeDir {
+					return fs.SkipDir
+				}
 			}
 			return nil
 		}
 
 		var requiredHeader *regexp.Regexp
 		switch {
+		case slices.Contains(excludeFiles, path):
```

Comment:
I moved this up because exact matches to the exclusions were still being tested for.